src/EnergyPlus/api/datatransfer.py

```diff
@@ -84,6 +84,33 @@ class DataExchange:
     outside calling code.
     """
 
+    class _APIDataEntry(Structure):
+        _fields_ = [("what", c_char_p),("name", c_char_p),("key", c_char_p),("type", c_char_p),]
+
+    class APIDataExchangePoint:
+        """
+        A class of string members that describe a single API exchange point.  The exchange described in this
+        class could represent output variables, output meters, actuators, and more.  The "type" member variable
+        can be used to filter a specific type.
+        """
+        def __init__(self, _what: str, _name: str, _key: str, _type: str):
+            #: This variable will hold the basic type of API data point, in string form.
+            #: This can be one of the following: "Actuator", "InternalVariable", "PluginGlobalVariable",
+            #: "PluginTrendVariable", "OutputMeter", or "OutputVariable".  Once the full list of data exchange points
+            #: are returned from a call to get_api_data, this can be used to quickly filter down to a specific type.
+            self.what = _what
+            #: This represents the name of the entry point, not the name of the specific instance of the entry point.
+            #: Some examples of this name could be "Chiller Heat Transfer Rate" -- which could be available for multiple
+            #: chillers.
+            self.name = _name
+            #: This represents the unique ID for this exchange point.  In the example of the chiller output variable,
+            #: this could be "Chiller 1". This is not used for meters
+            self.key = _key
+            #: This represents the "type" of exchange for this exchange point.  This is only used for actuators,
+            #: and represents the control actuation.  For a node setpoint actuation, this could be "temperature" or
+            #: "humidity", for example.
+            self.type = _type
```

Comment:
This class is the actual item returned in the Python API.  Same data as the C API, but in nice Python form, and nicely documented.